hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/validate/HazelcastSqlValidator.java

```diff
@@ -146,6 +150,42 @@ public void validateCall(SqlCall call, SqlValidatorScope scope) {
         super.validateCall(call, scope);
     }
 
+    @Override
+    public void validateQuery(SqlNode node, SqlValidatorScope scope, RelDataType targetRowType) {
+        super.validateQuery(node, scope, targetRowType);
+
+        if (node instanceof SqlSelect) {
+            // Derive the types for offset-fetch expressions, Calcite doesn't do
+            // that automatically.
+
+            SqlSelect select = (SqlSelect) node;
+
+            SqlNode offset = select.getOffset();
+            if (offset != null) {
+                deriveType(scope, offset);
+                validateNonNegativeValue(offset);
+            }
+
+            SqlNode fetch = select.getFetch();
+            if (fetch != null) {
+                deriveType(scope, fetch);
+                validateNonNegativeValue(fetch);
+            }
+        }
+    }
+
+    private void validateNonNegativeValue(SqlNode sqlNode) {
+        if (!(sqlNode instanceof SqlNumericLiteral)) {
+            throw error(sqlNode, "FETCH/OFFSET must be a numeric literal");
+        }
+        Object value = ((SqlNumericLiteral) sqlNode).getValue();
+        long value0 = ((Number) value).longValue();
+
+        if (value0 < 0L) {
+            throw error(sqlNode, "FETCH/OFFSET value cannot be negative: " + value0);
+        }
+    }
```

Comment:
it is just a note.
I run `testSelectFetchOffsetInvalid` in particular this test
```java
String sql1 = "SELECT " + intValField + " FROM " + stableMapName()
    + " OFFSET -5 ROWS FETCH FIRST 10 ROWS ONLY";

assertThrows(HazelcastSqlException.class, () -> assertSqlResultCount(sql1, 0));
``` 
It throws `HazelcastSqlException.class` with `calcite` message:
```
Encountered "-" at line 1, column 45.
Was expecting one of:
    <UNSIGNED_INTEGER_LITERAL> ...
    <APPROX_NUMERIC_LITERAL> ...
    <DECIMAL_NUMERIC_LITERAL> ...
    "?" ...
```
maybe we should make a note somewhere that currently we relay on `calcite` for negative and non-numeric `FETCH`/`OFFSET` values?